### PR TITLE
GameLab bug fix: Loading a project with animations that have the same name

### PR DESCRIPTION
**Pull request by caleybrock**
A couple of months ago we disallowed having animations with the same name in the UI, and then wrote a function that checks for duplicate names when loading a project and changes them if necessary. 

I got another report last week that a student project wasn't loading with this error:
screen shot 2017-03-22 at 3 39 19 pm
So I clearly missed some edge cases here (particularly with strings with `(` and `_` ) and with the order the duplicates are in. Wrote a basic test and the case that broke in the zendesk ticket and fixed how we check for duplicates.

### CODE REVIEW

File: apps/src/gamelab/animationListModule.js [hunk 1]
```diff
@@ -290,10 +294,13 @@ export function setInitialAnimationList(serializedAnimationList) {
   for (let i = 0; i < numberAnimations; i++) {
     const key = serializedAnimationList.orderedKeys[i];
     const name = serializedAnimationList.propsByKey[key].name;
-    for (let j = i + 1; j < numberAnimations; j++) {
+    for (let j = 0; j < numberAnimations; j++) {
+      if (j === i) {
+        continue;
+      }
```
Comment (ashercodeorg) [inline apps/src/gamelab/animationListModule.js]: Can this instead be done by starting the `j` counter at `i + 1`?

Comment (caleybrock) [reply apps/src/gamelab/animationListModule.js]: `continue` skips that iteration of the loop while `break` exits the loop so I don't think it's the same thing, but I don't think what I added is actually necessary so I'll update what you suggested.

File: apps/src/gamelab/animationListModule.js [hunk 2]
```diff
@@ -229,11 +229,15 @@ export function isNameUnique(name, animationListProps) {
 function generateAnimationName(baseName, animationList) {
   let unavailableNumbers = [];
   // Match names with the form baseName_#
-  const re = new RegExp(`^${baseName}_(\\d+)$`);
   for (let animation in animationList) {
-    let match = re.exec(animationList[animation].name);
-    if (match !== null) {
-      unavailableNumbers.push(parseInt(match[1]));
+    let animationName = animationList[animation].name;
+    if (animationName.startsWith(baseName)) {
```
Comment (Bjvanminnen) [inline apps/src/gamelab/animationListModule.js]: I would validate that this works in IE (which doesn't support startsWith natively). It's possible we'll need a polyfill.

File: apps/src/gamelab/animationListModule.js [hunk 3]
```diff
@@ -229,11 +229,15 @@ export function isNameUnique(name, animationListProps) {
...
+      animationName = animationName.replace(baseName, '');
+      if (animationName[0] === '_') {
+        const breakUpString = animationName.split('_');
```
Comment (Bjvanminnen) [inline apps/src/gamelab/animationListModule.js]: rather nitty, but i would describe this as a noun rather than a verb (i.e. brokenUpString). When I see verb, I think function.

File: apps/test/unit/gamelab/animationListModuleTest.js [hunk 4]
```diff
@@ -194,6 +194,29 @@ describe('animationListModule', function () {
       store.dispatch(setInitialAnimationList(animationList));
       expect(store.getState().animationTab.selectedAnimation).to.equal('');
     });
+
+    it('should not initial with multiple animations of the same name', function () {
```
Comment (Bjvanminnen) [inline apps/test/unit/gamelab/animationListModuleTest.js]: s/initial/initialize? (here and line 209)

**Comment by Bjvanminnen** [comment]
lgtm once you validate IE support

File: apps/src/gamelab/animationListModule.js [hunk 5]
```diff
@@ -231,11 +231,11 @@ function generateAnimationName(baseName, animationList) {
   // Match names with the form baseName_#
   for (let animation in animationList) {
     let animationName = animationList[animation].name;
-    if (animationName.startsWith(baseName)) {
+    if (animationName.lastIndexOf(baseName, 0) === 0) {
```
Comment (Bjvanminnen) [inline apps/src/gamelab/animationListModule.js]: Any reason this can't just be `indexOf`?

Could also do `animationName.substring(0, baseName.length) === baseName` I think.
